src/expr.cpp

```diff
@@ -894,9 +894,16 @@ std::shared_ptr<Expr> ArithmeticExpr::create(std::shared_ptr<PopulateCtx> ctx) {
     ctx->decArithDepth();
 
     if (ctx->getArithDepth() == 0) {
-        new_node->propagateType();
-        EvalCtx eval_ctx;
-        new_node->rebuild(eval_ctx);
+        Options &options = Options::getInstance();
+        bool allow_ub = !ctx->isTaken() &&
+                        (options.getAllowUBInDC() == OptionLevel::ALL ||
+                         (options.getAllowUBInDC() == OptionLevel::SOME &&
+                          rand_val_gen->getRandId(gen_pol->ub_in_dc_prob)));
+        if (!allow_ub) {
```

Comment:
Would be good to comment why this enables UB in the dead code.

Reply:
Fixed